apis/runtime/v1alpha1/api.proto

```diff
@@ -124,6 +124,9 @@ message ContainerResourceHookRequest {
   map<string, string> container_annotations = 3;
   LinuxContainerResources container_resources = 4;
   LinuxContainerResources pod_resources = 5;
+  // pod related annotations and labels
+  map<string, string> pod_annotations = 6;
+  map<string, string> pod_labels = 7;
```

Comment:
Why consider putting it at the interface level instead of letting the plugin get it by itself?

Reply:
> Why consider putting it at the interface level instead of letting the plugin get it by itself?
where to get pod info for plugin? talk with runtime-manager or containerd?

Reply:
> where to get pod info for plugin? talk with runtime-manager or containerd?

where to get pod info for plugin? talk with runtime-manager or containerd?

Reply:
> et. koordlet can get it from it local cache.

there may be only PreStartContainerHook registered, so koordlet could not catch pod level annotations/levels during RunSandbox stage without  PreRunPodSandboxHook hook.

yet, koordlet can register PreRunPodSandboxHook explicitly to solve, but we had better do not import dependency between two hooks(e.g PreRunPodSandboxHook and PreStartContainerHook).

Reply:
> > et. koordlet can get it from it local cache.
> 
> there may be only PreStartContainerHook registered, so koordlet could not catch pod level annotations/levels during RunSandbox stage without PreRunPodSandboxHook hook.
> 
> yet, koordlet can register PreRunPodSandboxHook explicitly to solve, but we had better do not import dependency between two hooks(e.g PreRunPodSandboxHook and PreStartContainerHook).

then， koordlet would be stateless, since no need to store the pod info

Reply:
1、pod syncing is for metric analysis and dynamic QoS policies, which is asynchronous with hook injections. So the injections will be ineffective if the hook wait

Reply:
Such as communication, keep these two fields, but let's do some deductions in the early stage(restricted to koordinator.sh)

Reply:
memo:
1、priority class for pod/container req
2、filter with DomainPrefix